torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,696 @@
+        # see: https://github.com/princeton-vl/RAFT-Stereo/blob/main/core/raft_stereo.py#L35-L37
+        self.shared_base = shared_base
+        self.residual_block: nn.Module = nn.Identity()
+        self.conv = nn.Conv2d(base_dim, output_dim, kernel_size=1)
+        if shared_base:
+            self.residual_block = block(base_dim, base_dim, norm_layer=nn.InstanceNorm2d, stride=1)
+            self.conv = nn.Conv2d(base_dim, output_dim, kernel_size=3, padding=1)
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.base_encoder(x)
+        x = self.residual_block(x)
+        x = self.conv(x)
+        return x
+
+
+class MultiLevelContextEncoder(nn.Module):
+    def __init__(
+        self,
+        base_encoder: nn.Module,
+        out_with_blocks: List[bool],
+        output_dim: int = 256,
+        block: Callable[..., nn.Module] = ResidualBlock,
+    ):
+        super().__init__()
+        self.num_level = len(out_with_blocks)
+        self.base_encoder = base_encoder
+        self.base_downsampling_ratio = base_encoder.downsampling_ratio
+        base_dim = base_encoder.output_dim
+
+        self.downsample_and_out_layers = nn.ModuleList(
+            [
+                nn.ModuleDict(
+                    {
+                        "downsampler": self._make_downsampler(block, base_dim, base_dim) if i > 0 else nn.Identity(),
+                        "out_hidden_state": self._make_out_layer(
+                            base_dim, output_dim // 2, with_block=out_with_blocks[i], block=block
+                        ),
+                        "out_context": self._make_out_layer(
+                            base_dim, output_dim // 2, with_block=out_with_blocks[i], block=block
+                        ),
+                    }
+                )
+                for i in range(self.num_level)
+            ]
+        )
+
+    def _make_out_layer(self, in_channels, out_channels, with_block=True, block=ResidualBlock):
+        block_layer = nn.Identity()
+        conv_layer = nn.Conv2d(in_channels, out_channels, kernel_size=3, padding=1)
+        if with_block:
+            block_layer = block(in_channels, in_channels, norm_layer=nn.BatchNorm2d, stride=1)
+        return nn.Sequential(block_layer, conv_layer)
+
+    def _make_downsampler(self, block, in_channels, out_channels):
+        block1 = block(in_channels, out_channels, norm_layer=nn.BatchNorm2d, stride=2)
+        block2 = block(out_channels, out_channels, norm_layer=nn.BatchNorm2d, stride=1)
+        return nn.Sequential(block1, block2)
+
+    def forward(self, x: Tensor) -> List[Tensor]:
+        x = self.base_encoder(x)
+        outs = []
+        for layer_dict in self.downsample_and_out_layers:
+            x = layer_dict["downsampler"](x)
+            outs.append(torch.cat([layer_dict["out_hidden_state"](x), layer_dict["out_context"](x)], dim=1))
+        return outs
+
+
+class ConvGRU(nn.Module):
```

Comment:
Here is the mypy error if I specify the typing:
```
raft_stereo.py:137: error: Signature of "forward" incompatible with supertype "ConvGRU"  [override]
        def forward(self, h: Tensor, context: List[Tensor], x: Tensor):
        ^
raft_stereo.py:137: note:      Superclass:
raft_stereo.py:137: note:          def forward(self, h: Any, x: Any) -> Any
raft_stereo.py:137: note:      Subclass:
raft_stereo.py:137: note:          def forward(self, h: Tensor, context: List[Tensor], x: Tensor) -> Any
Found 1 error in 1 file (checked 1 source file)
```
I have tried to put the `context` after `x` in the argument list, but it will still got the same error.
Also, here is a relevant discussion on this error: https://stackoverflow.com/questions/54346721/mypy-argument-of-method-incompatible-with-supertype

Reply:
This is solved by adding `# type: ignore[override]`